### PR TITLE
fetch builds by date range for a specific job

### PR DESCRIPTION
**Pull request by KevinRealAlcalde**

### CODE REVIEW

**Review by PierreBtz — COMMENTED** [review]
Won't +1 since I wrote it, it would be cheating, let's wait for an independent reviewer :)

File: fetch-builds-by-date-range-for-specific-job.groovy [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+def startDate = Instant.parse("2012-12-01T19:22:43.000Z")
+def endDate = Instant.now()
+def jobName = 'changeMe'
+
+println "Will return all the builds for job ${jobName} that ran between ${startDate} and ${endDate}"
+
+def builds = Jenkins.instance.getAllItems()
+        .findAll { it instanceof Job && it.fullName == jobName }
+        .collect { it.builds.byTimestamp(startDate.toEpochMilli(), endDate.toEpochMilli()) }
+        .flatten()
```
Comment (duemir) [inline fetch-builds-by-date-range-for-specific-job.groovy:15]:
```suggestion
def builds = Jenkins.get().getItemByFullName(jobName, hudson.model.Job).builds.byTimestamp(startDate.toEpochMilli(), endDate.toEpochMilli())
```

If a full name is expected, why bother with a search? Let us just get it.

**Review by duemir — CHANGES_REQUESTED** [review]
Why is this script necessary?
Who would use it?

File: fetch-builds-by-date-range-for-specific-job.groovy (hunk 1 above)
Comment (KevinRealAlcalde) [reply fetch-builds-by-date-range-for-specific-job.groovy:15]: Totally agree, thanks for the tip!

**Comment by KevinRealAlcalde** [comment]
I have had some customers requesting this information, and probably this repo is the best place for this script to be

**Review by Dohbedoh — APPROVED** [review]

**Pull request merged**